src/use_case/station_info/StationInfoInteractor.java

```diff
@@ -23,10 +23,17 @@ public void execute(StationInfoInputData stationInfoInputData) {
         // if not empty
         if (stationInfoDataAccessObject.incomingVehiclesNotEmpty(stationInfoInputData.getStationName())) {
 
-            // Creating a Station object using the station factory based on the name that the user input
-            Station station = stationInfoDataAccessObject.getStation(stationInfoInputData.getStationName());
+            //TODO: Review changes made from lines 27 - 31
```

Comment:
Looks clear and makes a lot of sense! Thanks for modifying, and I will make changes/add methods to relevant classes.